packages/react-components/react-combobox/src/utils/useComboboxBaseState.ts

```diff
@@ -56,13 +56,22 @@ export const useComboboxBaseState = (
       return props.defaultValue;
     }
 
+    const selectedOptionsText = getOptionsMatchingValue(optionValue => {
+      return selectedOptions.includes(optionValue);
+    }).map(option => option.text);
+
     if (multiselect) {
       // editable inputs should not display multiple selected options in the input as text
-      return editable ? '' : selectedOptions.join(', ');
+      return editable ? '' : selectedOptionsText.join(', ');
     }
 
-    return selectedOptions[0];
-  }, [controllableValue, editable, isFirstMount, multiselect, props.defaultValue, selectedOptions]);
+    return selectedOptionsText[0];
+
+    // do not change value after isFirstMount changes,
+    // we do not want to accidentally override defaultValue on a second render
+    // unless another value is intentionally set
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Discussed offline, but this use of `isFirstMount` (and disabling the `exhaustive-deps` warning) seem like they could lead to bugs. It might be worth rethinking how the value is calculated later, but it doesn't need to block this PR.

Reply:
i'm curious how this plays out in React 18 as well

Reply:
It actually seems like it works OK in React 18 + StrictMode: https://codesandbox.io/s/funny-bush-4vt1nm?file=/example.tsx

I can investigate more separate from this PR, though :)